components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -14723,4 +14724,87 @@ public String getUserID(String userName) throws APIManagementException {
         }
         return list;
     }
+
+    /**
+     * Add a new endpoint registry
+     *
+     * @param endpointRegistry EndpointRegistry
+     * @param tenantID  ID of the owner's tenant
+     * @return registryId
+     */
+    public String addEndpointRegistry(EndpointRegistry endpointRegistry, int tenantID) throws APIManagementException {
+        String query = SQLConstants.ADD_ENDPOINT_REGISTRY_SQL;
+        String uuid = UUID.randomUUID().toString();
+        try (Connection connection = APIMgtDBUtil.getConnection();
+             PreparedStatement ps = connection.prepareStatement(query)) {
+            ps.setString(1, uuid);
+            ps.setString(2, endpointRegistry.getName());
+            ps.setString(3, endpointRegistry.getType());
+            ps.setString(4, endpointRegistry.getMode());
+            ps.setInt(5, tenantID);
+            ps.setString(6, endpointRegistry.getOwner());
+            ps.setString(7, "");
```

Comment:
why empty?

Reply:
These db columns are for read_roles and write_roles. since, we planned to have registry roles per tenant, added empty values for now. 
Added a comment so we can update the values later based on the decision.